container-core/src/main/java/com/yahoo/container/di/CloudSubscriber.java

```diff
@@ -29,10 +32,16 @@ public class CloudSubscriber  implements Subscriber {
     // if waitNextGeneration has not yet been called, -1 should be returned
     private long generation = -1L;
 
-    CloudSubscriber(String name, ConfigSource configSource, Set<ConfigKey<ConfigInstance>> keys) {
+    CloudSubscriber(ExecutorService executor, String name, ConfigSource configSource, Set<ConfigKey<ConfigInstance>> keys) {
         this.name = name;
         this.subscriber = new ConfigSubscriber(configSource);
-        keys.forEach(k -> handles.put(k, subscriber.subscribe(k.getConfigClass(), k.getConfigId())));
+        Map<ConfigKey<ConfigInstance>, Future<ConfigHandle<ConfigInstance>>> futureHandles = new HashMap<>();
```

Comment:
Consider using vespajlib's `CompletableFutures.allOf()`

Reply:
Requires a rewrite that does seem straight forward.